pymc3/models/best.py

```diff
@@ -0,0 +1,168 @@
+"""
+Written by: Eric J. Ma
+Date: 11 November 2016
+Inspiration taken from many places, including the PyMC3 documentation.
+
+A note on the API design, for future contributors.
+
+There were some design choices made here for "default models" that may be
+modified in the future. I list the choices below:
+
+- A "model" is an object, like scikit-learn.
+- A model is instantiated with a DataFrame that houses the data.
+- Models accept other parameters as necessary.
+- Every model has a `.fit()` function that performs model fitting, like in
+  scikit-learn.
+- Every model has a `.plot_posterior()` function that returns a figure showing
+  the posterior distribution. Inspired by the pymc3 GLM module.
+- BEST uses ADVI, but this can (and should) be made an option; MCMC is also a
+  good tool to use.
+"""
+
+from ..distributions import StudentT, Exponential, Uniform, HalfCauchy
+from .. import Model
+from ..variational import advi, sample_vp
+import seaborn as sns
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+class BEST(object):
+    """BEST Model, based on Kruschke (2013).
+
+    Parameters
+    ----------
+    data : pandas DataFrame
+        A pandas dataframe which has the following data:
+        - Each row is one replicate measurement.
+        - There is a column that records the treatment name.
+        - There is a column that records the measured value for that replicate.
+
+    sample_col : str
+        The name of the column containing sample names.
+
+    output_col : str
+        The name of the column containing values to estimate.
+
+    baseline_name : str
+        The name of the "control" or "baseline".
+
+    Output
+    ------
+    model : PyMC3 model
+        Returns the BEST model containing
+    """
+    def __init__(self, data, sample_col, output_col, baseline_name):
+        super(BEST, self).__init__()
+        self.data = data
+        self.sample_col = sample_col
+        self.output_col = output_col
+        self.baseline_name = baseline_name
+        self.trace = None
+
+        self._convert_to_indices()
+
+    def _convert_to_indices(self):
+        """
+        Adds the "indices" column to self.data (DataFrame). This is necessary
+        for the simplified model specification in the "fit" function below.
+        """
+        sample_names = dict()
+        for i, name in enumerate(
+                list(np.unique(self.data[self.sample_col].values))):
+            print('Sample name {0} has the index {1}'.format(name, i))
+            sample_names[name] = i
+        self.data['indices'] = self.data[self.sample_col].apply(
+            lambda x: sample_names[x])
+
+    def fit(self, n_steps=500000):
```

Comment:
I plan to change this to use @twiecki's `sample_init` function when it's merged in. I'll then wrap the `kwargs`.